### PR TITLE
feat: Order Summary making similar to EventBrite

### LINKED ISSUE #2993: Order Summary Improvements and Making it similar to EventBrite
There is scope for several improvements in Tax Modal generated at `Order Checkout` -:
 
* [x] Include `Ticket Price` in row of every ticket
* [x] For a `Free Ticket` adding tax does not make any sense , Remove the `Column for Free Ticket`.
* [x] Fix the alignment issue in `Order Checkout`
* [x] Remove the `Discount Column` if no `Discount Code` is applied
* [ ] Show `Subtotal` inclusive of `Individual Ticket with Tax Rate`
* [x] Columns of `Payment Dialogue` are wrongly placed , Use standard order as in `Eventbrite` or any E-Commerce Site.

#### Screenshot -:

Screenshot 2019-05-23 at 4 19 45 PM

### PR DESCRIPTION
**Pull request by kushthedude**
Fixes #2993 except the ordering of columns

#### Changes Made in This PR -:

- Modifies the Summary Table for Free tickets or Discounted Tickets with Net Order Value Resulting to Zero.
- Hide Discount Column if discount is not applied .
- Subtotal includes the Tax for each ticket now 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by kushthedude** [comment]
@uds5501 @shreyanshdwivedi Please have a look

**Comment by uds5501** [comment]
@kushthedude screenshot?

**Comment by kushthedude** [comment]
### When Only Free Tickets Are Present

Screenshot 2019-06-07 at 12 26 44 AM

### When Due to Discount Net Amount is Zero

Screenshot 2019-06-07 at 12 28 46 AM

File: app/templates/components/orders/order-summary.hbs [hunk 1]
```diff
@@ -39,63 +39,94 @@
...
-          <th class="one wide">{{t 'Quantity'}}</th>
-          <th class="ui aligned two wide">{{t 'Subtotal'}}</th>
-          {{#if event.tax}}
-            <th class="four wide">{{t 'Tax'}}</th>
-          {{/if}}
-        </tr>
-      </thead>
-      <tbody>
-        {{#each tickets as |ticket|}}
+    {{#if (eq data.amount 0)}}
```
Comment (CosmicCoder96) [inline app/templates/components/orders/order-summary.hbs]: what if the amount is 0 BECAUSE of a discount code.

Comment (kushthedude) [reply app/templates/components/orders/order-summary.hbs]: Actually the second screenshot was for test purpose , How would it look and for Discount code with net amount zero , A column for `Discount` and `Ticket Price` would be shown without `Subtotal`

**Comment by kushthedude** [comment]
Screenshot 2019-06-07 at 9 30 51 AM
For Discounted Tickets

**Comment by kushthedude** [comment]
> What if there are free AND paid tickets in an order

For the case of Free and Paid, Before -: Price of Free was displayed as 0 and now Free would be written over the price tag

**Comment by kushthedude** [comment]
> The logic looks fine but it's too much code duplication.
> Is there any way you could clip it down?

@uds5501 If the changes looks fine , I will try to see if there is any more narrowing possible

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94  Please Review

**Comment by kushthedude** [comment]
@Anupam-dagar Actually the following build is not due to this PR , The build has been randomly failing since the refactor with the test which involves assert(login) , So not to worry

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 Review Please

File: app/templates/components/orders/order-summary.hbs [hunk 2]
```diff
@@ -39,63 +39,106 @@
...
+                <td>{{currency-symbol eventCurrency}} {{format-number ticket.price}}</td>
+                <td>{{currency-symbol eventCurrency}} {{format-number ticket.discount}}</td>
+              {{/if}}
+              <td>{{ticket-attendees data.attendees ticket.attendees}}</td>
+            </tr>
+          {{/each}}
+        </tbody>
+      </table>
+    {{else}}
+      <table class="ui very basic tablet stackable table order-summary center aligned">
```
Comment (shreyanshdwivedi) [inline app/templates/components/orders/order-summary.hbs]: @kushthedude there is a lot of duplicate lines from here onwards which can be easily reduced by changing the condition checks and using it at appropriate places. Can we change it and reduce the duplication of code?

Comment (kushthedude) [reply app/templates/components/orders/order-summary.hbs]: @shreyanshdwivedi I checked the lines again , If you actually see the code, the one which seems duplicate are mere condition and , I have tried to use less condition as much as possible , After the following there are mere two checks one about `Ticket Price` and one about `Ticket Discount` , I can even merge the checks but this wont be of any good as number of lines will be same . 
If you can find any duplication , Please report I will reduce them asap . Thanks

Comment (shreyanshdwivedi) [reply app/templates/components/orders/order-summary.hbs]: @kushthedude the lines 43-46 and lines 72-75 can be merged. The check on 47 and 77 can also be at one place. There are many such places which will reduce the duplication where check at line 42 can be used also.

Comment (kushthedude) [reply app/templates/components/orders/order-summary.hbs]: The check for 47 and 77 cant be at one place, As the one of 47 checks for 100 % discount as the net amount is zero and one at 77 checks for the if there is any discount and both will show different columns according to following

Comment (kushthedude) [reply app/templates/components/orders/order-summary.hbs]: @shreyanshdwivedi The one for 43-46 I really missed that Thanks for pointing

**Comment by kushthedude** [comment]
@shreyanshdwivedi Please have a look , I tried to inculcate some changes

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review please

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 Review Please

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 Review Please

**Comment by kushthedude** [comment]
@uds5501 @shreyanshdwivedi @mrsaicharan1 Please Review

**Comment by kushthedude** [comment]
@mrsaicharan1 @uds5501 @shreyanshdwivedi Have a look please

**Comment by kushthedude** [comment]
> Grand total is missing if discount code makes the ticket cost 0. Also in this case you should still show Grand Total as 0.

Making the following change

**Comment by kushthedude** [comment]
@mariobehling Please Check the following :
image

**Comment by kushthedude** [comment]
@CosmicCoder96 Review Please

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review Pleae

**Comment by kushthedude** [comment]
> There is a column "Subtotal" which disappeared. It should be there even if the price is 0. It should show that the price is actually zero.

@mariobehling , When a user only orders ‘Free Ticket’ , Eventbrite doesnt display any column except ticket type and quantity, i tried to replicate it in the following , if you say should i undo that case?

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 @mariobehling I have reduced the duplications as much as possible and also introduced the subtotal, Please review it

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review Please

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review Please

**Comment by kushthedude** [comment]
> Can you please the post the screenshot for the finalised changes right on top? I believe this PR has gone through a lot of iterations. Would be easy to review.

@mrsaicharan1 I will post ss of each and every case .

**Comment by mrsaicharan1** [comment]
@kushthedude okay.

**Comment by kushthedude** [comment]
Case 1 : When `Discount Code ` is applies and net amount : 0
image
Case 2 : When only Free tickets : 
image
Case 3: Paid ticket and no discount applied : 
image

@mrsaicharan1 Please check them .

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review Please

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review Please

**Comment by kushthedude** [comment]
@CosmicCoder96 
Here we go 👍 : 
- Case: 1
image
- Case: 2
image
- Case: 3
image
- Case: 4
image
- Case: 5
image

**Comment by CosmicCoder96** [comment]
@kushthedude wrap up this PR before the others, thanks!

**Comment by kushthedude** [comment]
@CosmicCoder96 Did the changes, You may notice a variety of `if` in table footer, It was just due to aligning `Grand Total` with `Sub Total` for each and every case

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 Please Review

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 Please Review

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 Please Review
--

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 Please Review
--

**Comment by kushthedude** [comment]
Close in favour of #3280 
--

**Pull request closed without merging**